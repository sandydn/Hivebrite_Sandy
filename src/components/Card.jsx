import React from 'react';
import styled from 'styled-components';

const CardStyle = styled.div`
    width: 215px;
    height: 100px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px;
    margin: 25px 45px 0px 3px;
    background: ${props => props.isSelected ? '#d2ebfa' : '#ebf1f5'};
    border: ${props => props.isSelected && '.5px solid #8baee6'};
    box-shadow: 2px 9px 20px rgba(0,0,0,0.1);
    cursor: pointer;
`;

const Column = styled.div`
    display: flex;
    flex-direction: column;
`;

const Item = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
`;

const CityName = styled.h2`
    margin: 5px 0px;
    font-size: 1.4em;
    width: 130px;
`;

const CityState = styled.h3`
    margin: 0px;
    text-transform: uppercase;
    font-size: .8em;
`;

const Icon = styled.img`
    width: 20px;
    height: 20px;
`;

const NumberItem = styled.h3`
    font-family: 'Advent Pro', sans-serif;
    font-size: 1em;
`;

const Card = props => {
    const {cityData: { 
        city, 
        state, 
        rank, 
        growth_from_2000_to_2013, 
        population 
    }, selectedCity, handleClick} = props;
     const isSelected = selectedCity.rank === rank ? true : false;

    return (
        <CardStyle onClick={() => handleClick(rank)} isSelected={isSelected}>
            <Column>
                <CityName>{city}</CityName>
                <CityState>{state}</CityState>
            </Column>
            <Column>
                <Item>
                    <Icon src="https://img.icons8.com/fluent/25/000000/conference-call.png"/>
                    <NumberItem>{population}</NumberItem>
                </Item>
                <Item>
                <Icon src={growth_from_2000_to_2013.includes('-') ? "https://img.icons8.com/fluent/35/000000/down.png" : "https://img.icons8.com/fluent/35/000000/up.png"}/>
                    <NumberItem>{growth_from_2000_to_2013}</NumberItem>
                </Item>
            </Column>
        </CardStyle>
    );
};

export default Card;