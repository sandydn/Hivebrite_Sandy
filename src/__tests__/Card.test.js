import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Card from '../components/Card.jsx';

describe('Card', () => {
    const onHandleClickMock = jest.fn();

    const props = {
        cityData: { 
            city: 'New York', 
            state: 'New York', 
            rank: '1', 
            growth_from_2000_to_2013: "4.8%", 
            population: "8405837" 
        }, 
        selectedCity: {
            city: "New York",
            growth_from_2000_to_2013: "4.8%",
            latitude: 40.7127837,
            longitude: -74.0059413,
            population: "8405837",
            rank: "1",
            state: "New York",
        },
        handleClick: () => onHandleClickMock('Arg')
    };

    const wrapper = shallow(<Card {...props} />);

    describe('Should pass on start.', () => {
        it('Should render correctly.', () => {
            const tree = renderer
                .create(<Card {...props} />)
                .toJSON();
        expect(tree).toMatchSnapshot();
        });

        it('Should call "onHandleClick" when clicking on the card.', () => {
            wrapper.find("[data-jest='card']").simulate('click');
        expect(onHandleClickMock).toHaveBeenCalled();
        });
    });
});