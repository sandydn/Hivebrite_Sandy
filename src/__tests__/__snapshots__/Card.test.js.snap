// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should pass on start. Should render correctly. 1`] = `
<div
  className="sc-AxjAm jgLRTK"
  data-jest="card"
  onClick={[Function]}
>
  <div
    className="sc-AxirZ eTxVFc"
  >
    <h2
      className="sc-AxhCb jAlGnI"
    >
      New York
    </h2>
    <h3
      className="sc-AxhUy jSGLIL"
    >
      New York
    </h3>
  </div>
  <div
    className="sc-AxirZ eTxVFc"
  >
    <div
      className="sc-AxiKw kKjxUC"
    >
      <img
        alt="population"
        className="sc-AxgMl jLPpZy"
        src="https://img.icons8.com/fluent/25/000000/conference-call.png"
      />
      <h3
        className="sc-AxheI kLPUmH"
      >
        8405837
      </h3>
    </div>
    <div
      className="sc-AxiKw kKjxUC"
    >
      <img
        className="sc-AxgMl jLPpZy"
        src="https://img.icons8.com/fluent/35/000000/up.png"
      />
      <h3
        className="sc-AxheI kLPUmH"
      >
        4.8%
      </h3>
    </div>
  </div>
</div>
`;
